plugins/functions/string.c

```diff
@@ -294,6 +304,165 @@ func_string_tokenize(grn_ctx *ctx, int n_args, grn_obj **args,
   return tokens;
 }
 
+#ifdef GRN_WITH_ONIGMO
+static grn_obj *
+string_regex_slice(grn_ctx *ctx, int n_args, grn_obj **args, grn_user_data *user_data)
+{
+  grn_obj *target, *pattern, *capture, *default_value, *result;
+  int64_t target_length, capture_name_length, capture_num = -1;
+  OnigRegex regex = NULL;
+  OnigPosition position, start, end;
+  OnigRegion *region;
+  const char *capture_name, *target_string;
+
+  if (!(n_args == 3 || n_args == 4)) {
+    GRN_PLUGIN_ERROR(ctx, GRN_INVALID_ARGUMENT,
+                     "[regex_slice] "
```

Comment:
Could you use `[string_slice]` here? Because this is an internal function of `string_slice()`. This feature is visible as `string_slice()` from users.